apis/apps/v1alpha1/opsrequest_types.go

```diff
@@ -448,11 +448,26 @@ type HorizontalScaling struct {
 	// +optional
 	Instances []InstanceTemplate `json:"instances,omitempty"`
 
-	// Specifies the names of instances to be scaled down.
-	// This provides control over which specific instances are targeted for termination when reducing the replica count.
+	// Scale out or in the specified instances based on the operator.
 	//
 	// +optional
 	OfflineInstances []string `json:"offlineInstances,omitempty"`
+
+	// Replicas will be automatically synchronized based on the offline instances when it is set to true and no any replicas are specified.
+	// +optional
+	AutoSyncReplicas bool `json:"autoSyncReplicas,omitempty"`
+
+	// Specifies the operator for this HorizontalScaling operation.
+	// Valid options are `Overwrite`, `Add`, and `Delete`.
+	//
+	// - `Overwrite`: Overwrite the replicas, instances, offlineInstances to the specified component. This is the default option.
+	// - `Add`: Create the specified number of replicas and instances, and scale in the specified instances.
+	//    This is the 'Add' operation at the data structure level of replicas, instances, and offlineInstances.
+	// - `Delete`: Delete the specified number of replicas and instances, and scale out the specified instances.
+	//    This is the 'Delete' operation at the data structure level of replicas, instances, and offlineInstances.
+	// +kubebuilder:default=Overwrite
+	// +optional
+	Operator HScaleOperator `json:"operator"`
```

Comment:
// Specifies the operator to modify `offlineInstances` during horizontal scaling.
//
// Valid values:
// - "Overwrite" (default): Replaces `offlineInstances` with the specified instances.
// - "Add": Appends the specified instances to `offlineInstances`.
// - "Delete": Removes the specified instances from `offlineInstances`.

Reply:
the scope of the operator is replicas, instances and offlineInstances.